packages/mongo-nav/src/MongoNav.tsx

```diff
@@ -18,12 +18,104 @@ import {
   MongoNavInterface,
   OrganizationInterface,
   ProjectInterface,
+  HostsInterface,
 } from './types';
 
 const ErrorCodeMap: Record<number, ErrorCode> = {
   401: ErrorCode.NO_AUTHORIZATION,
 } as const;
 
+type UseMongoNavData = Pick<
+  MongoNavInterface,
+  | 'mode'
+  | 'activeOrgId'
+  | 'activeProjectId'
+  | 'loadData'
+  | 'dataFixtures'
+  | 'onSuccess'
+  | 'onError'
+> & {
+  hosts: Required<NonNullable<MongoNavInterface['hosts']>>;
+};
+
+function useMongoNavData({
```

Comment:
Really really like breaking this out. Much more easy to reason about independent from the rest of the component.